torch/csrc/distributed/c10d/comm.cpp

```diff
@@ -79,4 +80,29 @@ void broadcast_coalesced(
   }
 }
 
+GradBucket::GradBucket(std::vector<at::Tensor>& tensors) : tensors_(tensors){};
+
+std::vector<at::Tensor> GradBucket::get_tensors() {
+  return tensors_;
+};
+
+PythonCommHook::PythonCommHook(py::object state, py::object hook)
+    : state_(std::move(state)), hook_(std::move(hook)){};
+c10::intrusive_ptr<torch::jit::Future> PythonCommHook::operate(
+    const GradBucket& bucket) {
+  py::gil_scoped_acquire acquire;
+
+  c10::intrusive_ptr<torch::jit::Future> fut;
```

Comment:
why do we need this Future object here?

Reply:
Frankly I was just following the reference casting in `PythonFutureWrapper`. There is no matching functional-style cast from `std::shared_ptr<torch::jit::PythonFutureWrapper>` to `c10::intrusive_ptr<torch::jit::Future>`. Therefore, the following is not allowed:
```
  return c10::intrusive_ptr<torch::jit::Future>(hook_(state_, bucket)
      .cast<std::shared_ptr<torch::jit::PythonFutureWrapper>>());
```
Please let me know if I'm missing something here and there is a better way of doing this without `fut`.